app/scenes/Settings/Security.js

```diff
@@ -18,39 +18,51 @@ function Security() {
   const team = useCurrentTeam();
   const { t } = useTranslation();
   const { showToast } = useToasts();
-  const [sharing, setSharing] = useState(team.documentEmbeds);
-  const [documentEmbeds, setDocumentEmbeds] = useState(team.guestSignin);
-  const [guestSignin, setGuestSignin] = useState(team.sharing);
+  const [sharing, setSharing] = useState(team.sharing);
+  const [documentEmbeds, setDocumentEmbeds] = useState(team.documentEmbeds);
+  const [guestSignin, setGuestSignin] = useState(team.guestSignin);
 
-  const showSuccessMessage = debounce(() => {
-    showToast(t("Settings saved"), { type: "success" });
-  }, 500);
+  const showSuccessMessage = React.useCallback(
+    debounce(() => {
+      showToast(t("Settings saved"), { type: "success" });
+    }, 250),
+    []
+  );
 
-  const handleChange = React.useCallback(
-    async (ev: SyntheticInputEvent<*>) => {
-      switch (ev.target.name) {
-        case "sharing":
-          setSharing(ev.target.checked);
-          break;
-        case "documentEmbeds":
-          setDocumentEmbeds(ev.target.checked);
-          break;
-        case "guestSignin":
-          setGuestSignin(ev.target.checked);
-          break;
-        default:
-      }
+  const handleChange = React.useCallback(async (ev: SyntheticInputEvent<*>) => {
+    switch (ev.target.name) {
+      case "sharing":
+        setSharing(ev.target.checked);
+        break;
+      case "documentEmbeds":
+        setDocumentEmbeds(ev.target.checked);
+        break;
+      case "guestSignin":
+        setGuestSignin(ev.target.checked);
+        break;
+      default:
+    }
+  }, []);
 
+  React.useEffect(() => {
+    async function save() {
       await auth.updateTeam({
```

Comment:
Not possible to use the variables immediately after `setState` as they are not updated, moved to a `useEffect` hook instead